extension/src/popup/helpers/formatters.ts

```diff
@@ -82,3 +82,15 @@ export const formatAmount = (
     newCursor: cursorPosition + commaDiff - cleanedDiff,
   };
 };
+
+export const formatAmount = (val: string) => {
+  const decimal = new Intl.NumberFormat("en-US", { style: "decimal" });
+  const [wholeVal, remainderVal] = val.split(".");
+  const formattedWholeVal = decimal.format(Number(wholeVal)).toString();
+
+  if (remainderVal) {
+    return `${formattedWholeVal}.${remainderVal}`;
+  }
+
+  return formattedWholeVal;
```

Comment:
I think you can use this in `formatAmountPreserveCursor` above, right?

Reply:
Felt risky to me to try to combine these since one targets user input and accounts for the cursor positioning while the other formats read only display values. I could take another look at combining these.